### ISSUE TITLE
Build issues because `BitmapAnim` is an incomplete type 

### ISSUE THREAD

**Issue by shtrophic** [wolfpld/vv#4]
Hello, I'm getting a build error on 3.0:

<details>
<summary>cmake</summary>

```
mkdir build
cd build
cmake -GNinja ..
```

```
-- The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "2.3.0")
-- CPM: Adding package tracy@ (master)
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- TRACY_ENABLE: OFF
-- TRACY_ON_DEMAND: OFF
-- TRACY_CALLSTACK: OFF
-- TRACY_NO_CALLSTACK: OFF
-- TRACY_NO_CALLSTACK_INLINES: OFF
-- TRACY_ONLY_LOCALHOST: OFF
-- TRACY_NO_BROADCAST: OFF
-- TRACY_ONLY_IPV4: OFF
-- TRACY_NO_CODE_TRANSFER: OFF
-- TRACY_NO_CONTEXT_SWITCH: OFF
-- TRACY_NO_EXIT: OFF
-- TRACY_NO_SAMPLING: OFF
-- TRACY_NO_VERIFY: OFF
-- TRACY_NO_VSYNC_CAPTURE: OFF
-- TRACY_NO_FRAME_IMAGE: OFF
-- TRACY_NO_SYSTEM_TRACING: OFF
-- TRACY_PATCHABLE_NOPSLEDS: OFF
-- TRACY_DELAYED_INIT: OFF
-- TRACY_MANUAL_LIFETIME: OFF
-- TRACY_FIBERS: OFF
-- TRACY_NO_CRASH_HANDLER: OFF
-- TRACY_TIMER_FALLBACK: OFF
-- TRACY_LIBUNWIND_BACKTRACE: OFF
-- TRACY_SYMBOL_OFFLINE_RESOLVE: OFF
-- TRACY_LIBBACKTRACE_ELF_DYNLOAD_SUPPORT: OFF
-- TRACY_DEBUGINFOD: OFF
-- TRACY_VERBOSE: OFF
-- TRACY_DEMANGLE: OFF
-- CPM: Adding package stb@0 (2e2bef463a5b53ddf8bb788e25da6b8506314c08)
-- Populating stb
-- Configuring done (0.1s)
-- Generating done (0.0s)
-- Build files have been written to: /home/chris/Source/vv/build/_deps/stb-subbuild
[1/9] Creating directories for 'stb-populate'
[1/9] Performing download step (git clone) for 'stb-populate'
[2/9] Performing update step for 'stb-populate'
-- Already at requested ref: 2e2bef463a5b53ddf8bb788e25da6b8506314c08
[3/9] No patch step for 'stb-populate'
[5/9] No configure step for 'stb-populate'
[6/9] No build step for 'stb-populate'
[7/9] No install step for 'stb-populate'
[8/9] No test step for 'stb-populate'
[9/9] Completed 'stb-populate'
-- CPM: Adding package base64@0.5.2 (v0.5.2)
-- /home/chris/Source/vv/build/_deps/base64-src
-- Looking for getopt.h
-- Looking for getopt.h - found
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- The following packages have been found:

 * Threads
 * OpenMP
   Allows to utilize OpenMP

-- The following features have been enabled:

 * SSSE3, add SSSE 3 codepath
 * SSE4.1, add SSE 4.1 codepath
 * SSE4.2, add SSE 4.2 codepath
 * AVX, add AVX codepath
 * AVX2, add AVX2 codepath
 * AVX512, add AVX512 codepath

-- The following features have been disabled:

 * CLI, enables the CLI executable for encoding and decoding
 * OpenMP codec, spreads codec work accross multiple threads
 * NEON32, add NEON32 codepath
 * NEON64, add NEON64 codepath

-- Checking for module 'cairo'
--   Found cairo, version 1.18.2
-- Checking for module 'OpenEXR'
--   Found OpenEXR, version 3.3.2
-- Checking for module 'libheif'
--   Found libheif, version 1.19.5
-- Checking for module 'libjpeg'
--   Found libjpeg, version 3.0.4
-- Checking for modules 'libjxl;libjxl_threads'
--   Found libjxl, version 0.11.1
--   Found libjxl_threads, version 0.11.1
-- Checking for module 'lcms2'
--   Found lcms2, version 2.16
-- Checking for module 'libpng'
--   Found libpng, version 1.6.44
-- Checking for module 'libraw'
--   Found libraw, version 0.21.3
-- Checking for module 'librsvg-2.0'
--   Found librsvg-2.0, version 2.59.2
-- Checking for module 'libsixel'
--   Found libsixel, version 1.10.3
-- Checking for module 'libtiff-4'
--   Found libtiff-4, version 4.7.0
-- Checking for module 'libwebpdemux'
--   Found libwebpdemux, version 1.4.0
-- Checking for module 'zlib'
--   Found zlib, version 1.3.1
-- Configuring done (64.7s)
-- Generating done (0.1s)
-- Build files have been written to: /home/chris/Source/vv/build
```
</details>

<details>
<summary>build</summary>

```
[1/72] Scanning /home/chris/Source/vv/src/util/Bitmap.cpp for CXX dependencies
[2/72] Scanning /home/chris/Source/vv/src/util/Logs.cpp for CXX dependencies
[3/72] Scanning /home/chris/Source/vv/src/util/Home.cpp for CXX dependencies
[4/72] Scanning /home/chris/Source/vv/src/util/BitmapAnim.cpp for CXX dependencies
[5/72] Scanning /home/chris/Source/vv/src/util/Callstack.cpp for CXX dependencies
[6/72] Scanning /home/chris/Source/vv/src/image/DdsLoader.cpp for CXX dependencies
[7/72] Scanning /home/chris/Source/vv/src/image/JpgLoader.cpp for CXX dependencies
[8/72] Scanning /home/chris/Source/vv/src/util/BitmapHdr.cpp for CXX dependencies
[9/72] Scanning /home/chris/Source/vv/src/util/FileBuffer.cpp for CXX dependencies
[10/72] Scanning /home/chris/Source/vv/src/image/PcxLoader.cpp for CXX dependencies
[11/72] Scanning /home/chris/Source/vv/src/image/ImageLoader.cpp for CXX dependencies
[12/72] Scanning /home/chris/Source/vv/src/image/HeifLoader.cpp for CXX dependencies
[13/72] Scanning /home/chris/Source/vv/src/image/PvrLoader.cpp for CXX dependencies
[14/72] Scanning /home/chris/Source/vv/src/image/PngLoader.cpp for CXX dependencies
[15/72] Scanning /home/chris/Source/vv/src/image/JxlLoader.cpp for CXX dependencies
[16/72] Scanning /home/chris/Source/vv/src/image/TiffLoader.cpp for CXX dependencies
[17/72] Scanning /home/chris/Source/vv/src/image/WebpLoader.cpp for CXX dependencies
[18/72] Scanning /home/chris/Source/vv/src/image/ExrLoader.cpp for CXX dependencies
[19/72] Scanning /home/chris/Source/vv/src/util/stb_image_resize_impl.cpp for CXX dependencies
[20/72] Building CXX object _deps/tracy-build/CMakeFiles/TracyClient.dir/public/TracyClient.cpp.o
[21/72] Generating CXX dyndep file CMakeFiles/mcoreutil.dir/CXX.dd
[22/72] Building C object _deps/base64-build/CMakeFiles/base64.dir/lib/lib.c.o
[23/72] Scanning /home/chris/Source/vv/src/image/RawLoader.cpp for CXX dependencies
[24/72] Scanning /home/chris/Source/vv/src/image/vector/PdfImage.cpp for CXX dependencies
[25/72] Linking CXX static library _deps/tracy-build/libTracyClient.a
[26/72] Scanning /home/chris/Source/vv/src/tools/vv/Terminal.cpp for CXX dependencies
[27/72] Scanning /home/chris/Source/vv/src/image/StbImageLoader.cpp for CXX dependencies
[28/72] Building C object CMakeFiles/mcoreimage.dir/src/image/bcdec.c.o
[29/72] Building CXX object CMakeFiles/mcoreutil.dir/src/util/Bitmap.cpp.o
[30/72] Building C object _deps/base64-build/CMakeFiles/base64.dir/lib/codec_choose.c.o
[31/72] Scanning /home/chris/Source/vv/src/tools/vv/vv.cpp for CXX dependencies
[32/72] Building C object _deps/base64-build/CMakeFiles/base64.dir/lib/tables/tables.c.o
[33/72] Generating CXX dyndep file CMakeFiles/vv.dir/CXX.dd
[34/72] Scanning /home/chris/Source/vv/src/image/vector/SvgImage.cpp for CXX dependencies
[35/72] Generating CXX dyndep file CMakeFiles/mcoreimage.dir/CXX.dd
[36/72] Building C object _deps/base64-build/CMakeFiles/base64.dir/lib/arch/generic/codec.c.o
[37/72] Building C object _deps/base64-build/CMakeFiles/base64.dir/lib/arch/ssse3/codec.c.o
[38/72] Building CXX object CMakeFiles/mcoreutil.dir/src/util/Logs.cpp.o
[39/72] Building CXX object CMakeFiles/mcoreutil.dir/src/util/Home.cpp.o
[40/72] Building CXX object CMakeFiles/mcoreimage.dir/src/image/DdsLoader.cpp.o
FAILED: CMakeFiles/mcoreimage.dir/src/image/DdsLoader.cpp.o 
/usr/bin/c++ -DDISABLE_CALLSTACK -I/home/chris/Source/vv/build -I/home/chris/Source/vv -I/home/chris/Source/vv/src -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/pixman-1 -I/usr/include/OpenEXR -I/usr/include/Imath -I/usr/include/webp -I/usr/include/libraw -I/usr/lib/pkgconfig/../../include/librsvg-2.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libxml2 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/home/chris/Source/vv/build/_deps/stb-src -isystem /home/chris/Source/vv/build/_deps/tracy-src/public -std=gnu++20 -fdiagnostics-color=always -MD -MT CMakeFiles/mcoreimage.dir/src/image/DdsLoader.cpp.o -MF CMakeFiles/mcoreimage.dir/src/image/DdsLoader.cpp.o.d -fmodules-ts -fmodule-mapper=CMakeFiles/mcoreimage.dir/src/image/DdsLoader.cpp.o.modmap -MD -fdeps-format=p1689r5 -x c++ -o CMakeFiles/mcoreimage.dir/src/image/DdsLoader.cpp.o -c /home/chris/Source/vv/src/image/DdsLoader.cpp
In Datei, eingebunden von /usr/include/c++/14.2.1/memory:78,
                 von /home/chris/Source/vv/src/image/ImageLoader.hpp:3,
                 von /home/chris/Source/vv/src/image/DdsLoader.hpp:5,
                 von /home/chris/Source/vv/src/image/DdsLoader.cpp:4:
/usr/include/c++/14.2.1/bits/unique_ptr.h: In Instanziierung von »void std::default_delete<_Tp>::operator()(_Tp*) const [mit _Tp = BitmapAnim]«:
/usr/include/c++/14.2.1/bits/unique_ptr.h:398:17:   erfordert durch »std::unique_ptr<_Tp, _Dp>::~unique_ptr() [mit _Tp = BitmapAnim; _Dp = std::default_delete<BitmapAnim>]«
  398 |           get_deleter()(std::move(__ptr));
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/chris/Source/vv/src/image/ImageLoader.hpp:26:75:   von hier erfordert
   26 |     [[nodiscard]] virtual std::unique_ptr<BitmapAnim> LoadAnim() { return nullptr; }
      |                                                                           ^~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h:91:23: Fehler: Ungültige Anwendung von »sizeof« auf unvollständigen Typen »BitmapAnim«
   91 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h: In Instanziierung von »void std::default_delete<_Tp>::operator()(_Tp*) const [mit _Tp = BitmapHdr]«:
/usr/include/c++/14.2.1/bits/unique_ptr.h:398:17:   erfordert durch »std::unique_ptr<_Tp, _Dp>::~unique_ptr() [mit _Tp = BitmapHdr; _Dp = std::default_delete<BitmapHdr>]«
  398 |           get_deleter()(std::move(__ptr));
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/chris/Source/vv/src/image/ImageLoader.hpp:27:73:   von hier erfordert
   27 |     [[nodiscard]] virtual std::unique_ptr<BitmapHdr> LoadHdr() { return nullptr; }
      |                                                                         ^~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h:91:23: Fehler: Ungültige Anwendung von »sizeof« auf unvollständigen Typen »BitmapHdr«
   91 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
[41/72] Building CXX object CMakeFiles/mcoreutil.dir/src/util/BitmapAnim.cpp.o
[42/72] Building CXX object CMakeFiles/mcoreutil.dir/src/util/BitmapHdr.cpp.o
[43/72] Building CXX object CMakeFiles/mcoreutil.dir/src/util/Callstack.cpp.o
[44/72] Building CXX object CMakeFiles/mcoreimage.dir/src/image/JpgLoader.cpp.o
FAILED: CMakeFiles/mcoreimage.dir/src/image/JpgLoader.cpp.o 
/usr/bin/c++ -DDISABLE_CALLSTACK -I/home/chris/Source/vv/build -I/home/chris/Source/vv -I/home/chris/Source/vv/src -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/pixman-1 -I/usr/include/OpenEXR -I/usr/include/Imath -I/usr/include/webp -I/usr/include/libraw -I/usr/lib/pkgconfig/../../include/librsvg-2.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libxml2 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/home/chris/Source/vv/build/_deps/stb-src -isystem /home/chris/Source/vv/build/_deps/tracy-src/public -std=gnu++20 -fdiagnostics-color=always -MD -MT CMakeFiles/mcoreimage.dir/src/image/JpgLoader.cpp.o -MF CMakeFiles/mcoreimage.dir/src/image/JpgLoader.cpp.o.d -fmodules-ts -fmodule-mapper=CMakeFiles/mcoreimage.dir/src/image/JpgLoader.cpp.o.modmap -MD -fdeps-format=p1689r5 -x c++ -o CMakeFiles/mcoreimage.dir/src/image/JpgLoader.cpp.o -c /home/chris/Source/vv/src/image/JpgLoader.cpp
In Datei, eingebunden von /usr/include/c++/14.2.1/memory:78,
                 von /home/chris/Source/vv/src/image/ImageLoader.hpp:3,
                 von /home/chris/Source/vv/src/image/JpgLoader.hpp:3,
                 von /home/chris/Source/vv/src/image/JpgLoader.cpp:6:
/usr/include/c++/14.2.1/bits/unique_ptr.h: In Instanziierung von »void std::default_delete<_Tp>::operator()(_Tp*) const [mit _Tp = BitmapAnim]«:
/usr/include/c++/14.2.1/bits/unique_ptr.h:398:17:   erfordert durch »std::unique_ptr<_Tp, _Dp>::~unique_ptr() [mit _Tp = BitmapAnim; _Dp = std::default_delete<BitmapAnim>]«
  398 |           get_deleter()(std::move(__ptr));
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/chris/Source/vv/src/image/ImageLoader.hpp:26:75:   von hier erfordert
   26 |     [[nodiscard]] virtual std::unique_ptr<BitmapAnim> LoadAnim() { return nullptr; }
      |                                                                           ^~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h:91:23: Fehler: Ungültige Anwendung von »sizeof« auf unvollständigen Typen »BitmapAnim«
   91 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h: In Instanziierung von »void std::default_delete<_Tp>::operator()(_Tp*) const [mit _Tp = BitmapHdr]«:
/usr/include/c++/14.2.1/bits/unique_ptr.h:398:17:   erfordert durch »std::unique_ptr<_Tp, _Dp>::~unique_ptr() [mit _Tp = BitmapHdr; _Dp = std::default_delete<BitmapHdr>]«
  398 |           get_deleter()(std::move(__ptr));
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/chris/Source/vv/src/image/ImageLoader.hpp:27:73:   von hier erfordert
   27 |     [[nodiscard]] virtual std::unique_ptr<BitmapHdr> LoadHdr() { return nullptr; }
      |                                                                         ^~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h:91:23: Fehler: Ungültige Anwendung von »sizeof« auf unvollständigen Typen »BitmapHdr«
   91 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
[45/72] Building CXX object CMakeFiles/mcoreimage.dir/src/image/HeifLoader.cpp.o
FAILED: CMakeFiles/mcoreimage.dir/src/image/HeifLoader.cpp.o 
/usr/bin/c++ -DDISABLE_CALLSTACK -I/home/chris/Source/vv/build -I/home/chris/Source/vv -I/home/chris/Source/vv/src -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/pixman-1 -I/usr/include/OpenEXR -I/usr/include/Imath -I/usr/include/webp -I/usr/include/libraw -I/usr/lib/pkgconfig/../../include/librsvg-2.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libxml2 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/home/chris/Source/vv/build/_deps/stb-src -isystem /home/chris/Source/vv/build/_deps/tracy-src/public -std=gnu++20 -fdiagnostics-color=always -MD -MT CMakeFiles/mcoreimage.dir/src/image/HeifLoader.cpp.o -MF CMakeFiles/mcoreimage.dir/src/image/HeifLoader.cpp.o.d -fmodules-ts -fmodule-mapper=CMakeFiles/mcoreimage.dir/src/image/HeifLoader.cpp.o.modmap -MD -fdeps-format=p1689r5 -x c++ -o CMakeFiles/mcoreimage.dir/src/image/HeifLoader.cpp.o -c /home/chris/Source/vv/src/image/HeifLoader.cpp
In Datei, eingebunden von /usr/include/c++/14.2.1/memory:78,
                 von /home/chris/Source/vv/src/image/HeifLoader.hpp:3,
                 von /home/chris/Source/vv/src/image/HeifLoader.cpp:7:
/usr/include/c++/14.2.1/bits/unique_ptr.h: In Instanziierung von »void std::default_delete<_Tp>::operator()(_Tp*) const [mit _Tp = BitmapAnim]«:
/usr/include/c++/14.2.1/bits/unique_ptr.h:398:17:   erfordert durch »std::unique_ptr<_Tp, _Dp>::~unique_ptr() [mit _Tp = BitmapAnim; _Dp = std::default_delete<BitmapAnim>]«
  398 |           get_deleter()(std::move(__ptr));
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/chris/Source/vv/src/image/ImageLoader.hpp:26:75:   von hier erfordert
   26 |     [[nodiscard]] virtual std::unique_ptr<BitmapAnim> LoadAnim() { return nullptr; }
      |                                                                           ^~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h:91:23: Fehler: Ungültige Anwendung von »sizeof« auf unvollständigen Typen »BitmapAnim«
   91 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
[46/72] Building CXX object CMakeFiles/mcoreimage.dir/src/image/ImageLoader.cpp.o
FAILED: CMakeFiles/mcoreimage.dir/src/image/ImageLoader.cpp.o 
/usr/bin/c++ -DDISABLE_CALLSTACK -I/home/chris/Source/vv/build -I/home/chris/Source/vv -I/home/chris/Source/vv/src -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/pixman-1 -I/usr/include/OpenEXR -I/usr/include/Imath -I/usr/include/webp -I/usr/include/libraw -I/usr/lib/pkgconfig/../../include/librsvg-2.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libxml2 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/home/chris/Source/vv/build/_deps/stb-src -isystem /home/chris/Source/vv/build/_deps/tracy-src/public -std=gnu++20 -fdiagnostics-color=always -MD -MT CMakeFiles/mcoreimage.dir/src/image/ImageLoader.cpp.o -MF CMakeFiles/mcoreimage.dir/src/image/ImageLoader.cpp.o.d -fmodules-ts -fmodule-mapper=CMakeFiles/mcoreimage.dir/src/image/ImageLoader.cpp.o.modmap -MD -fdeps-format=p1689r5 -x c++ -o CMakeFiles/mcoreimage.dir/src/image/ImageLoader.cpp.o -c /home/chris/Source/vv/src/image/ImageLoader.cpp
In Datei, eingebunden von /usr/include/c++/14.2.1/memory:78,
                 von /home/chris/Source/vv/src/image/ImageLoader.hpp:3,
                 von /home/chris/Source/vv/src/image/DdsLoader.hpp:5,
                 von /home/chris/Source/vv/src/image/ImageLoader.cpp:4:
/usr/include/c++/14.2.1/bits/unique_ptr.h: In Instanziierung von »void std::default_delete<_Tp>::operator()(_Tp*) const [mit _Tp = BitmapAnim]«:
/usr/include/c++/14.2.1/bits/unique_ptr.h:398:17:   erfordert durch »std::unique_ptr<_Tp, _Dp>::~unique_ptr() [mit _Tp = BitmapAnim; _Dp = std::default_delete<BitmapAnim>]«
  398 |           get_deleter()(std::move(__ptr));
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/chris/Source/vv/src/image/ImageLoader.hpp:26:75:   von hier erfordert
   26 |     [[nodiscard]] virtual std::unique_ptr<BitmapAnim> LoadAnim() { return nullptr; }
      |                                                                           ^~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h:91:23: Fehler: Ungültige Anwendung von »sizeof« auf unvollständigen Typen »BitmapAnim«
   91 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h: In Instanziierung von »void std::default_delete<_Tp>::operator()(_Tp*) const [mit _Tp = BitmapHdr]«:
/usr/include/c++/14.2.1/bits/unique_ptr.h:398:17:   erfordert durch »std::unique_ptr<_Tp, _Dp>::~unique_ptr() [mit _Tp = BitmapHdr; _Dp = std::default_delete<BitmapHdr>]«
  398 |           get_deleter()(std::move(__ptr));
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/chris/Source/vv/src/image/ImageLoader.hpp:27:73:   von hier erfordert
   27 |     [[nodiscard]] virtual std::unique_ptr<BitmapHdr> LoadHdr() { return nullptr; }
      |                                                                         ^~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h:91:23: Fehler: Ungültige Anwendung von »sizeof« auf unvollständigen Typen »BitmapHdr«
   91 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
[47/72] Building CXX object CMakeFiles/mcoreutil.dir/src/util/FileBuffer.cpp.o
[48/72] Building CXX object CMakeFiles/mcoreutil.dir/src/util/stb_image_resize_impl.cpp.o
[49/72] Building CXX object CMakeFiles/mcoreimage.dir/src/image/ExrLoader.cpp.o
FAILED: CMakeFiles/mcoreimage.dir/src/image/ExrLoader.cpp.o 
/usr/bin/c++ -DDISABLE_CALLSTACK -I/home/chris/Source/vv/build -I/home/chris/Source/vv -I/home/chris/Source/vv/src -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/pixman-1 -I/usr/include/OpenEXR -I/usr/include/Imath -I/usr/include/webp -I/usr/include/libraw -I/usr/lib/pkgconfig/../../include/librsvg-2.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libxml2 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/home/chris/Source/vv/build/_deps/stb-src -isystem /home/chris/Source/vv/build/_deps/tracy-src/public -std=gnu++20 -fdiagnostics-color=always -MD -MT CMakeFiles/mcoreimage.dir/src/image/ExrLoader.cpp.o -MF CMakeFiles/mcoreimage.dir/src/image/ExrLoader.cpp.o.d -fmodules-ts -fmodule-mapper=CMakeFiles/mcoreimage.dir/src/image/ExrLoader.cpp.o.modmap -MD -fdeps-format=p1689r5 -x c++ -o CMakeFiles/mcoreimage.dir/src/image/ExrLoader.cpp.o -c /home/chris/Source/vv/src/image/ExrLoader.cpp
In Datei, eingebunden von /usr/include/OpenEXR/ImfFrameBuffer.h:16,
                 von /usr/include/OpenEXR/ImfRgbaFile.h:21,
                 von /home/chris/Source/vv/src/image/ExrLoader.cpp:4:
/usr/include/OpenEXR/ImfForward.h:25:51: Warnung: Typattribute ignoriert, nachdem Typ bereits definiert ist [-Wattributes]
   25 | struct IMF_EXPORT_TYPE                            Chromaticities;
      |                                                   ^~~~~~~~~~~~~~
In Datei, eingebunden von /usr/include/c++/14.2.1/memory:78,
                 von /home/chris/Source/vv/src/image/ExrLoader.hpp:3,
                 von /home/chris/Source/vv/src/image/ExrLoader.cpp:8:
/usr/include/c++/14.2.1/bits/unique_ptr.h: In Instanziierung von »void std::default_delete<_Tp>::operator()(_Tp*) const [mit _Tp = BitmapAnim]«:
/usr/include/c++/14.2.1/bits/unique_ptr.h:398:17:   erfordert durch »std::unique_ptr<_Tp, _Dp>::~unique_ptr() [mit _Tp = BitmapAnim; _Dp = std::default_delete<BitmapAnim>]«
  398 |           get_deleter()(std::move(__ptr));
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/chris/Source/vv/src/image/ImageLoader.hpp:26:75:   von hier erfordert
   26 |     [[nodiscard]] virtual std::unique_ptr<BitmapAnim> LoadAnim() { return nullptr; }
      |                                                                           ^~~~~~~
/usr/include/c++/14.2.1/bits/unique_ptr.h:91:23: Fehler: Ungültige Anwendung von »sizeof« auf unvollständigen Typen »BitmapAnim«
   91 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

</details>

**Comment by wolfpld** [comment]
Fixed on `v3.0.1`.

**Comment by shtrophic** [comment]
Confirmed, thanks!
